examples/supplementary/arg_iter/src/lib.rs

```diff
@@ -0,0 +1,164 @@
+#![feature(rustc_private)]
+#![warn(unused_extern_crates)]
+
+extern crate rustc_hir;
+extern crate rustc_middle;
+extern crate rustc_span;
+
+use clippy_utils::diagnostics::span_lint_and_help;
+use rustc_hir::{Body, FnDecl, intravisit::FnKind};
+use rustc_lint::{LateContext, LateLintPass};
+use rustc_middle::ty;
+use rustc_span::{
+    Span,
+    def_id::{DefId, LocalDefId},
+};
+
+dylint_linting::declare_late_lint! {
+    pub ARG_ITER,
```

Comment:
Could you please add doc comments here to match your README?

A good example can be seen in the recent PR #1623: https://github.com/trailofbits/dylint/pull/1623/files#diff-910f3f4e06c9a9d6a262f36213e71630d42f1749c01f2b6d979596fabef2fcfcR18-R93

Note that there is a script [update_example_READMEs.sh](https://github.com/trailofbits/dylint/blob/master/scripts/update_example_READMEs.sh) that generates READMEs from the examples' doc comments. Ideally, the doc comments you add would be such that your existing README is generated from them.